### PR TITLE
feat: add mdx support to storybook

### LINKED ISSUE #185: Docs & MDX Support
I would like to write a few words about our new components so people understand how and why you will use it. That's available in storybook through DocsPages.  It either extracts comments from a component or you start with mdx in the beginning, basically writing markdown content mixed with your components mark up.

See the [Docs Overview](https://github.com/storybookjs/storybook/tree/next/addons/docs) and the [MDX description](https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/mdx.md).

Right now my comments above components are not extracted. Maybe someone knows if this is already supported out of the box and then let's discuss if we would benefit from mdx.

### PR DESCRIPTION
**Pull request by hofmann-felix**
Resolves: https://github.com/satellytes/satellytes.com/issues/185
Storybook: https://deploy-preview-200--satellytes-website-storybook.netlify.app
## What's included?
- [Storybook Docs ](https://github.com/storybookjs/storybook/blob/next/addons/docs/README.md) was added and configured. This allows us to add a mdx documentation to each existing story. 
- A mdx documentation was added to the teaser story. This documentation still needs to be improved, currently it only serves to show the functionality of the plugin. 

## Notes
There are several ways to add documentation: Currently in the Teaser component, the existing story is kept (`teaser.stories.tsx`) and only the documentation (`teaser.stories.tsx`) comes from the mdx file. This allows (as before) the pure component to be displayed under "Canvas" and the mdx documentation under "Docs". 
Bildschirmfoto 2021-11-22 um 14 46 33
Instructions and further options can be found here:
- [How to write CSF stories with MDX docs](https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/recipes.md#csf-stories-with-mdx-docs)
- [General story book mdx documentation](https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/mdx.md)

## Preview
Bildschirmfoto 2021-11-22 um 14 30 14

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+  stories: [
+    '../src/**/*.stories.mdx',
+    '../src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
+  ],
   addons: [
     '@storybook/addon-links',
     '@storybook/addon-essentials',
     'storybook-addon-gatsby',
     'storybook-react-i18next',
+    '@storybook/addon-docs',
```
Comment (fischeversenker) [inline .storybook/main.js]: According to the [documentation](https://storybook.js.org/docs/react/writing-docs/introduction) and the [npm package](https://www.npmjs.com/package/@storybook/addon-essentials) this is already part of `@storybook/addon-essentials`. Are you sure we need this explicitly (both in this fiel as well as a dependency in the `package.json`)?

**Review by feedm3 — APPROVED** [review]
One step closer to a very cool component library 👍

File: src/components/teasers/teaser.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+# Teaser
+
+Teasers are used to link to another page and give the user a brief overview of the content of that page.
+This requires a `headline`, the path to the page (`linkTo`) and a short text, which is entered as a child.
+In addition, an illustration or an image (`cover`), a formatted date (`dateFormatted`) and a `topline` can be displayed.
+
+
+
+## Teaser Layouts
```
Comment (fischeversenker) [inline src/components/teasers/teaser.stories.mdx]: Those are a\
\
\
lot of new lines. Do we need them?

File: src/components/teasers/teaser.stories.tsx [hunk 3]
```diff
@@ -1,16 +1,11 @@
...
 import { ILLUSTRATION_NAMES } from '../illustration/illustration-set';
 
-export default {
-  component: Teaser,
-  title: 'Components/Teaser',
-} as ComponentMeta<typeof Teaser>;
-
 const Template: ComponentStory<typeof Teaser> = (args) => <Teaser {...args} />;
 
 export const Regular = Template.bind({});
```
Comment (fischeversenker) [inline src/components/teasers/teaser.stories.tsx:11]: Even though it's not part of this PR: Would you mind changing the default value for the `title` property in these stories to `Teaser title` instead of `Leadbox title`?

**Review by fischeversenker — COMMENTED** [review]
Thanks for taking care of this. It works nicely.

I do wonder if we need to explicitly state `addon-docs` as a dependency. According to the documentation it's already part of `addon-essentials`. Did you check?
I can imagine that this PR does not need to touch the storybook config, the `package.json` nor the `yarn.lock`.

File: .storybook/main.js (hunk 1 above)
Comment (hofmann-felix) [reply .storybook/main.js]: Yes you are definitely right. It is already included in `addon-essentials`. I removed the changes

File: src/components/teasers/teaser.stories.tsx (hunk 3 above)
Comment (hofmann-felix) [reply src/components/teasers/teaser.stories.tsx:11]: I've already changed it in https://github.com/satellytes/satellytes.com/pull/198. Thanks anyway 👍

**Review by fischeversenker — APPROVED** [review]
Thanks for the cleanup. Looks great now :raised_hands:
